sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DoNotCallExitMethods.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Windows.Forms;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        public void Foo()
+        {
+            Environment.Exit(0); // Noncompliant

```

Comment:
You forgot the cases specified in the exclude (i.e. the rule should not report when called in the main method).